src/Tribe/Utils/Theme_Compatibility.php

```diff
@@ -0,0 +1,266 @@
+<?php
+/**
+ * Add theme compatibility classes.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Utils
+ */
+namespace Tribe\Utils;
+
+use Compatibility_Classes;
+use \WP_Theme;
+
+class Theme_Compatibility {
+	/**
+	 * List of themes which have compatibility requirements.
+	 *
+	 * @since TBD
+	 *
+	 * @var   array
+	 */
+	protected static $themes = [
+		'avada',
+		'divi',
+		'enfold',
+		'genesis',
+		'twentyseventeen',
+		'twentynineteen',
+		'twentytwenty',
+		'twentytwentyone',
+	];
+
+	/**
+	 * Checks if the current theme needs a compatibility fix.
+	 *
+	 * @since  TBD
+	 *
+	 * @param string $theme Optionally, pass a specific theme name in to see if compatibility
+	 *                      is required for that theme.
+	 *
+	 * @return boolean
+	 */
+	public static function is_compatibility_required( $theme = null ) {
+		// Passing a theme name skips these checks.
+		if ( empty( $theme ) ) {
+			$current_theme = static::get_current_theme( true );
+
+			if ( empty( $current_theme ) || empty( $current_theme->get_template() ) ) {
+				return false;
+			}
+
+			$theme = $current_theme->get_template();
+		}
+
+		$required = in_array( $theme, static::get_registered_themes() );
+
+		return tribe_is_truthy( apply_filters( 'tribe_compatibility_required', $required, $theme ) );
```

Comment:
Docblock.

Reply:
@Camwyn looks like we're still missing this